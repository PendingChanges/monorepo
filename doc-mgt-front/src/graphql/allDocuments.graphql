query allDocuments($skip: Int, $take: Int) {
  allDocuments(skip:$skip, take: $take) {
    items {
      extension
      fileNameWithoutExtension
      id
      name
      version {
        major
        minor
      }
    }
    pageInfo{
      hasNextPage
      hasPreviousPage      
    }
    totalCount
  }
}