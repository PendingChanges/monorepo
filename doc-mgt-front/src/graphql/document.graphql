query document($id: UUID!, $version: String) {
  document(id: $id, version: $version) {
    extension
    fileNameWithoutExtension
    id
    name
    version {
      major
      minor
    }
  }
}
