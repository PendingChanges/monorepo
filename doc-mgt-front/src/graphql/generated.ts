import { gql } from 'apollo-angular';
import { Injectable } from '@angular/core';
import * as Apollo from 'apollo-angular';
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: { input: string; output: string; }
  String: { input: string; output: string; }
  Boolean: { input: boolean; output: boolean; }
  Int: { input: number; output: number; }
  Float: { input: number; output: number; }
  /** The `Short` scalar type represents non-fractional signed whole 16-bit numeric values. Short can represent values between -(2^15) and 2^15 - 1. */
  Short: { input: any; output: any; }
  UUID: { input: any; output: any; }
};

/** A segment of a collection. */
export type AllDocumentsCollectionSegment = {
  __typename?: 'AllDocumentsCollectionSegment';
  /** A flattened list of the items. */
  items?: Maybe<Array<Document>>;
  /** Information to aid in pagination. */
  pageInfo: CollectionSegmentInfo;
  totalCount: Scalars['Int']['output'];
};

export enum ApplyPolicy {
  AfterResolver = 'AFTER_RESOLVER',
  BeforeResolver = 'BEFORE_RESOLVER',
  Validation = 'VALIDATION'
}

/** Information about the offset pagination. */
export type CollectionSegmentInfo = {
  __typename?: 'CollectionSegmentInfo';
  /** Indicates whether more items exist following the set defined by the clients arguments. */
  hasNextPage: Scalars['Boolean']['output'];
  /** Indicates whether more items exist prior the set defined by the clients arguments. */
  hasPreviousPage: Scalars['Boolean']['output'];
};

export type Document = {
  __typename?: 'Document';
  extension: Scalars['String']['output'];
  fileNameWithoutExtension: Scalars['String']['output'];
  id: Scalars['UUID']['output'];
  name: Scalars['String']['output'];
  version: Version;
};

export type Query = {
  __typename?: 'Query';
  allDocuments?: Maybe<AllDocumentsCollectionSegment>;
  document?: Maybe<Document>;
};


export type QueryAllDocumentsArgs = {
  skip?: InputMaybe<Scalars['Int']['input']>;
  sortBy?: InputMaybe<Scalars['String']['input']>;
  sortDirection?: InputMaybe<Scalars['String']['input']>;
  take?: InputMaybe<Scalars['Int']['input']>;
};


export type QueryDocumentArgs = {
  id: Scalars['UUID']['input'];
  version?: InputMaybe<Scalars['String']['input']>;
};

export type Version = {
  __typename?: 'Version';
  build: Scalars['Int']['output'];
  major: Scalars['Int']['output'];
  majorRevision: Scalars['Short']['output'];
  minor: Scalars['Int']['output'];
  minorRevision: Scalars['Short']['output'];
  revision: Scalars['Int']['output'];
};

export type AllDocumentsQueryVariables = Exact<{
  skip?: InputMaybe<Scalars['Int']['input']>;
  take?: InputMaybe<Scalars['Int']['input']>;
}>;


export type AllDocumentsQuery = { __typename?: 'Query', allDocuments?: { __typename?: 'AllDocumentsCollectionSegment', totalCount: number, items?: Array<{ __typename?: 'Document', extension: string, fileNameWithoutExtension: string, id: any, name: string, version: { __typename?: 'Version', major: number, minor: number } }> | null, pageInfo: { __typename?: 'CollectionSegmentInfo', hasNextPage: boolean, hasPreviousPage: boolean } } | null };

export type DocumentQueryVariables = Exact<{
  id: Scalars['UUID']['input'];
  version?: InputMaybe<Scalars['String']['input']>;
}>;


export type DocumentQuery = { __typename?: 'Query', document?: { __typename?: 'Document', extension: string, fileNameWithoutExtension: string, id: any, name: string, version: { __typename?: 'Version', major: number, minor: number } } | null };

export const AllDocumentsDocument = gql`
    query allDocuments($skip: Int, $take: Int) {
  allDocuments(skip: $skip, take: $take) {
    items {
      extension
      fileNameWithoutExtension
      id
      name
      version {
        major
        minor
      }
    }
    pageInfo {
      hasNextPage
      hasPreviousPage
    }
    totalCount
  }
}
    `;

  @Injectable({
    providedIn: 'root'
  })
  export class AllDocumentsGQL extends Apollo.Query<AllDocumentsQuery, AllDocumentsQueryVariables> {
    document = AllDocumentsDocument;
    
    constructor(apollo: Apollo.Apollo) {
      super(apollo);
    }
  }
export const DocumentDocument = gql`
    query document($id: UUID!, $version: String) {
  document(id: $id, version: $version) {
    extension
    fileNameWithoutExtension
    id
    name
    version {
      major
      minor
    }
  }
}
    `;

  @Injectable({
    providedIn: 'root'
  })
  export class DocumentGQL extends Apollo.Query<DocumentQuery, DocumentQueryVariables> {
    document = DocumentDocument;
    
    constructor(apollo: Apollo.Apollo) {
      super(apollo);
    }
  }