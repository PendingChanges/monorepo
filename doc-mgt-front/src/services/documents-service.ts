import { Injectable } from '@angular/core';
import { ApolloQueryResult } from '@apollo/client/core';
import {
  AllDocumentsCollectionSegment,
  AllDocumentsGQL,
  AllDocumentsQuery,
  AllDocumentsQueryVariables,
  DocumentGQL,
  DocumentQuery,
} from '../graphql/generated';
import { map, Observable, Subject, tap } from 'rxjs';
import { QueryRef } from 'apollo-angular';

@Injectable({
  providedIn: 'root',
})
export class DocumentsService {
  private _allDocumentsQueryRef: QueryRef<
    AllDocumentsQuery,
    AllDocumentsQueryVariables
  >;

  public allDocuments$: Observable<AllDocumentsCollectionSegment>;

  public allDocumentsLoading$: Subject<boolean> = new Subject<boolean>();

  constructor(
    private _documentsGQL: AllDocumentsGQL,
    private _documentGQL: DocumentGQL
  ) {
    this._allDocumentsQueryRef = this._documentsGQL.watch({
      skip: 0,
      take: 15,
    });

    this.allDocuments$ = this._allDocumentsQueryRef.valueChanges.pipe(
      tap(() => this.allDocumentsLoading$.next(false)),
      map((result) => <AllDocumentsCollectionSegment>result.data.allDocuments)
    );
  }

  public refreshDocuments(skip: number, take: number): void {
    this.allDocumentsLoading$.next(true);
    this._allDocumentsQueryRef.refetch({
      skip: skip,
      take: take,
    });
  }

  public getDocument(id: string): Observable<ApolloQueryResult<DocumentQuery>> {
    return this._documentGQL.watch({ id }).valueChanges;
  }
}
