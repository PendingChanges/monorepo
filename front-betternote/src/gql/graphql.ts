/* eslint-disable */
import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: { input: string; output: string; }
  String: { input: string; output: string; }
  Boolean: { input: boolean; output: boolean; }
  Int: { input: number; output: number; }
  Float: { input: number; output: number; }
  UUID: { input: any; output: any; }
};

/** A segment of a collection. */
export type AllSubjectsCollectionSegment = {
  __typename?: 'AllSubjectsCollectionSegment';
  /** A flattened list of the items. */
  items?: Maybe<Array<Subject>>;
  /** Information to aid in pagination. */
  pageInfo: CollectionSegmentInfo;
  totalCount: Scalars['Int']['output'];
};

/** Defines when a policy shall be executed. */
export enum ApplyPolicy {
  /** After the resolver was executed. */
  AfterResolver = 'AFTER_RESOLVER',
  /** Before the resolver was executed. */
  BeforeResolver = 'BEFORE_RESOLVER',
  /** The policy is applied in the validation step before the execution. */
  Validation = 'VALIDATION'
}

/** Information about the offset pagination. */
export type CollectionSegmentInfo = {
  __typename?: 'CollectionSegmentInfo';
  /** Indicates whether more items exist following the set defined by the clients arguments. */
  hasNextPage: Scalars['Boolean']['output'];
  /** Indicates whether more items exist prior the set defined by the clients arguments. */
  hasPreviousPage: Scalars['Boolean']['output'];
};

export type CreateSubjectInput = {
  description: Scalars['String']['input'];
  tagsId?: InputMaybe<Array<Scalars['UUID']['input']>>;
  title: Scalars['String']['input'];
};

export type CreateTagInput = {
  value: Scalars['String']['input'];
};

export type Mutation = {
  __typename?: 'Mutation';
  createSubject: SubjectCreatedPayload;
  createTag: TagCreatedPayload;
  deleteTag: Scalars['UUID']['output'];
};


export type MutationCreateSubjectArgs = {
  input: CreateSubjectInput;
};


export type MutationCreateTagArgs = {
  input: CreateTagInput;
};


export type MutationDeleteTagArgs = {
  id: Scalars['UUID']['input'];
};

export type Query = {
  __typename?: 'Query';
  allSubjects?: Maybe<AllSubjectsCollectionSegment>;
  allTags: Array<Tag>;
};


export type QueryAllSubjectsArgs = {
  skip?: InputMaybe<Scalars['Int']['input']>;
  sortBy?: InputMaybe<Scalars['String']['input']>;
  sortDirection?: InputMaybe<Scalars['String']['input']>;
  take?: InputMaybe<Scalars['Int']['input']>;
};

export type Subject = {
  __typename?: 'Subject';
  description: Scalars['String']['output'];
  id: Scalars['UUID']['output'];
  title: Scalars['String']['output'];
};

export type SubjectCreatedPayload = {
  __typename?: 'SubjectCreatedPayload';
  id: Scalars['UUID']['output'];
};

export type Tag = {
  __typename?: 'Tag';
  id: Scalars['UUID']['output'];
  value: Scalars['String']['output'];
};

export type TagCreatedPayload = {
  __typename?: 'TagCreatedPayload';
  id: Scalars['UUID']['output'];
};

export type AllSubjectsQueryVariables = Exact<{ [key: string]: never; }>;


export type AllSubjectsQuery = { __typename?: 'Query', allSubjects?: { __typename?: 'AllSubjectsCollectionSegment', totalCount: number, items?: Array<{ __typename?: 'Subject', id: any, title: string, description: string }> | null, pageInfo: { __typename?: 'CollectionSegmentInfo', hasNextPage: boolean, hasPreviousPage: boolean } } | null };


export const AllSubjectsDocument = {"kind":"Document","definitions":[{"kind":"OperationDefinition","operation":"query","name":{"kind":"Name","value":"allSubjects"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"allSubjects"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"items"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"id"}},{"kind":"Field","name":{"kind":"Name","value":"title"}},{"kind":"Field","name":{"kind":"Name","value":"description"}}]}},{"kind":"Field","name":{"kind":"Name","value":"pageInfo"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"hasNextPage"}},{"kind":"Field","name":{"kind":"Name","value":"hasPreviousPage"}}]}},{"kind":"Field","name":{"kind":"Name","value":"totalCount"}}]}}]}}]} as unknown as DocumentNode<AllSubjectsQuery, AllSubjectsQueryVariables>;