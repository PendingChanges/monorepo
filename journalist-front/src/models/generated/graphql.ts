export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  DateTime: any;
};

/** A segment of a collection. */
export type AllClientsCollectionSegment = {
  __typename?: 'AllClientsCollectionSegment';
  /** A flattened list of the items. */
  items?: Maybe<Array<Client>>;
  /** Information to aid in pagination. */
  pageInfo: CollectionSegmentInfo;
  totalCount: Scalars['Int'];
};

/** A segment of a collection. */
export type AllIdeasCollectionSegment = {
  __typename?: 'AllIdeasCollectionSegment';
  /** A flattened list of the items. */
  items?: Maybe<Array<Idea>>;
  /** Information to aid in pagination. */
  pageInfo: CollectionSegmentInfo;
  totalCount: Scalars['Int'];
};

/** A segment of a collection. */
export type AllPitchesCollectionSegment = {
  __typename?: 'AllPitchesCollectionSegment';
  /** A flattened list of the items. */
  items?: Maybe<Array<Pitch>>;
  /** Information to aid in pagination. */
  pageInfo: CollectionSegmentInfo;
  totalCount: Scalars['Int'];
};

export enum ApplyPolicy {
  AfterResolver = 'AFTER_RESOLVER',
  BeforeResolver = 'BEFORE_RESOLVER',
  Validation = 'VALIDATION'
}

export type Client = {
  __typename?: 'Client';
  id: Scalars['String'];
  name: Scalars['String'];
  nbOfPitches: Scalars['Int'];
  userId: Scalars['String'];
};

export type ClientAddedPayload = {
  __typename?: 'ClientAddedPayload';
  clientId?: Maybe<Scalars['String']>;
};

/** Information about the offset pagination. */
export type CollectionSegmentInfo = {
  __typename?: 'CollectionSegmentInfo';
  /** Indicates whether more items exist following the set defined by the clients arguments. */
  hasNextPage: Scalars['Boolean'];
  /** Indicates whether more items exist prior the set defined by the clients arguments. */
  hasPreviousPage: Scalars['Boolean'];
};

export type CreateClientInput = {
  name: Scalars['String'];
};

export type CreateIdeaInput = {
  description?: InputMaybe<Scalars['String']>;
  name: Scalars['String'];
};

export type CreatePitchInput = {
  clientId: Scalars['String'];
  content: PitchContentInput;
  deadLineDate?: InputMaybe<Scalars['DateTime']>;
  ideaId: Scalars['String'];
  issueDate?: InputMaybe<Scalars['DateTime']>;
};

export type DeleteClientInput = {
  id: Scalars['String'];
};

export type DeleteIdeaInput = {
  id: Scalars['String'];
};

export type DeletePitchInput = {
  id: Scalars['String'];
};

export type DomainError = {
  __typename?: 'DomainError';
  domainErrors: Array<Error>;
  message: Scalars['String'];
};

export type Error = {
  __typename?: 'Error';
  code: Scalars['String'];
  label: Scalars['String'];
};

export type Idea = {
  __typename?: 'Idea';
  description?: Maybe<Scalars['String']>;
  id: Scalars['String'];
  name: Scalars['String'];
  nbOfPitches: Scalars['Int'];
  userId: Scalars['String'];
};

export type IdeaAddedPayload = {
  __typename?: 'IdeaAddedPayload';
  ideaId?: Maybe<Scalars['String']>;
};

export type ModifyIdeaInput = {
  id: Scalars['String'];
  newDescription?: InputMaybe<Scalars['String']>;
  newName: Scalars['String'];
};

export type ModifyPitchInput = {
  clientId: Scalars['String'];
  content: PitchContentInput;
  deadLineDate?: InputMaybe<Scalars['DateTime']>;
  id: Scalars['String'];
  ideaId: Scalars['String'];
  issueDate?: InputMaybe<Scalars['DateTime']>;
};

export type Mutation = {
  __typename?: 'Mutation';
  addClient: ClientAddedPayload;
  addIdea: IdeaAddedPayload;
  addPitch: PitchAddedPayload;
  modifyIdea: Scalars['String'];
  modifyPitch: Scalars['String'];
  removeClient: Scalars['String'];
  removeIdea: Scalars['String'];
  removePitch: Scalars['String'];
  renameClient: Scalars['String'];
};


export type MutationAddClientArgs = {
  createClient: CreateClientInput;
};


export type MutationAddIdeaArgs = {
  createIdea: CreateIdeaInput;
};


export type MutationAddPitchArgs = {
  createPitch: CreatePitchInput;
};


export type MutationModifyIdeaArgs = {
  modifyIdea: ModifyIdeaInput;
};


export type MutationModifyPitchArgs = {
  modifyPitch: ModifyPitchInput;
};


export type MutationRemoveClientArgs = {
  deleteClient: DeleteClientInput;
};


export type MutationRemoveIdeaArgs = {
  deleteIdea: DeleteIdeaInput;
};


export type MutationRemovePitchArgs = {
  deletePitch: DeletePitchInput;
};


export type MutationRenameClientArgs = {
  renameClient: RenameClientInput;
};

export type Pitch = {
  __typename?: 'Pitch';
  client?: Maybe<Client>;
  clientId: Scalars['String'];
  content: PitchContent;
  deadLineDate?: Maybe<Scalars['DateTime']>;
  id: Scalars['String'];
  idea?: Maybe<Idea>;
  ideaId: Scalars['String'];
  issueDate?: Maybe<Scalars['DateTime']>;
  userId: Scalars['String'];
};

export type PitchAddedPayload = {
  __typename?: 'PitchAddedPayload';
  pitchId?: Maybe<Scalars['String']>;
};

export type PitchContent = {
  __typename?: 'PitchContent';
  summary?: Maybe<Scalars['String']>;
  title: Scalars['String'];
};

export type PitchContentInput = {
  summary?: InputMaybe<Scalars['String']>;
  title: Scalars['String'];
};

export type Query = {
  __typename?: 'Query';
  allClients?: Maybe<AllClientsCollectionSegment>;
  allIdeas?: Maybe<AllIdeasCollectionSegment>;
  allPitches?: Maybe<AllPitchesCollectionSegment>;
  autoCompleteClient: Array<Client>;
  autoCompleteIdea: Array<Idea>;
  client?: Maybe<Client>;
  idea?: Maybe<Idea>;
  pitch?: Maybe<Pitch>;
};


export type QueryAllClientsArgs = {
  skip?: InputMaybe<Scalars['Int']>;
  sortBy?: InputMaybe<Scalars['String']>;
  sortDirection?: InputMaybe<Scalars['String']>;
  take?: InputMaybe<Scalars['Int']>;
};


export type QueryAllIdeasArgs = {
  skip?: InputMaybe<Scalars['Int']>;
  sortBy?: InputMaybe<Scalars['String']>;
  sortDirection?: InputMaybe<Scalars['String']>;
  take?: InputMaybe<Scalars['Int']>;
};


export type QueryAllPitchesArgs = {
  clientId?: InputMaybe<Scalars['String']>;
  ideaId?: InputMaybe<Scalars['String']>;
  skip?: InputMaybe<Scalars['Int']>;
  sortBy?: InputMaybe<Scalars['String']>;
  sortDirection?: InputMaybe<Scalars['String']>;
  take?: InputMaybe<Scalars['Int']>;
};


export type QueryAutoCompleteClientArgs = {
  text: Scalars['String'];
};


export type QueryAutoCompleteIdeaArgs = {
  text: Scalars['String'];
};


export type QueryClientArgs = {
  id: Scalars['String'];
};


export type QueryIdeaArgs = {
  id: Scalars['String'];
};


export type QueryPitchArgs = {
  id: Scalars['String'];
};

export type RenameClientInput = {
  id: Scalars['String'];
  newName: Scalars['String'];
};
