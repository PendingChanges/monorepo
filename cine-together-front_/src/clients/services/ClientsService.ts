import { Injectable } from '@angular/core';
import { MutationResult, QueryRef } from 'apollo-angular';
import { map, Observable } from 'rxjs';
import {
  AllClientsCollectionSegment,
  Client,
  ClientAddedPayload,
  DeleteClientInput,
  QueryAllClientsArgs,
  QueryAutoCompleteClientArgs,
  RenameClientInput,
} from 'src/models/generated/graphql';
import { AddClientMutation } from 'src/clients/mutations/AddClientMutation';
import { CreateClientInput } from 'src/models/generated/graphql';
import { AllClientsQuery } from 'src/clients/queries/AllClientsQuery';
import { AutoCompleteClientQuery } from 'src/clients/queries/AutoCompleteClientQuery';
import { ApolloQueryResult } from '@apollo/client/core';
import { ClientQuery } from '../queries/ClientQuery';
import { DeleteClientMutation } from '../mutations/DeleteClientMutation';
import { RenameClientMutation } from '../mutations/RenameClientMutation';

@Injectable({
  providedIn: 'root',
})
export class ClientsService {
  private _allClientsQueryRef: QueryRef<
    { allClients: AllClientsCollectionSegment },
    QueryAllClientsArgs
  > | null = null;

  constructor(
    private _allClientsQuery: AllClientsQuery,
    private _clientQuery: ClientQuery,
    private _addClientMutation: AddClientMutation,
    private _deleteClientMutation: DeleteClientMutation,
    private _autoCompleteClientQuery: AutoCompleteClientQuery,
    private _renameClientMutation: RenameClientMutation
  ) {}

  public getClients(args: QueryAllClientsArgs) {
    return this._allClientsQuery.fetch(args);
  }

  public getClient(id: string) {
    return this._clientQuery.watch({
      id: id,
    }).valueChanges;
  }

  public addClient(
    value: CreateClientInput
  ): Observable<MutationResult<{ addClient: ClientAddedPayload }>> {
    return this._addClientMutation.mutate(value);
  }

  public renameClient(
    value: RenameClientInput
  ): Observable<MutationResult<string>> {
    return this._renameClientMutation.mutate(value);
  }

  public removeClient(
    deleteClientInput: DeleteClientInput
  ): Observable<MutationResult<{ removeClient: string }>> {
    return this._deleteClientMutation.mutate(deleteClientInput);
  }

  public autoComplete(text: string): Observable<Client[]> {
    return this._autoCompleteClientQuery
      .fetch(<QueryAutoCompleteClientArgs>{ text: text })
      .pipe(map((result) => result.data.autoCompleteClient));
  }
}
