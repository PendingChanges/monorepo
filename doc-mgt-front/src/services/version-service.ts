import { Injectable } from '@angular/core';
import { Version } from '../graphql/generated';

@Injectable({
  providedIn: 'root',
})
export class VersionService {
  public getFormatedVersion(version: Version | undefined): string {
    return version ? `${version.major}.${version.minor}` : '';
  }
}
