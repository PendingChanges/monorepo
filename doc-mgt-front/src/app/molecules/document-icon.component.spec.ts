import { ComponentFixture, TestBed } from '@angular/core/testing';
import { DocumentIconComponent } from './document-icon.component';
import { DocumentDocument } from '../../services/generated';
import { Document } from '../../graphql/generated';
import { ComponentRef } from '@angular/core';

describe('DocumentIconComponent', () => {
  let component: DocumentIconComponent;
  let conponentRef: ComponentRef<DocumentIconComponent>;
  let fixture: ComponentFixture<DocumentIconComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [DocumentIconComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(DocumentIconComponent);
    component = fixture.componentInstance;
    conponentRef = fixture.componentRef;
    fixture.detectChanges();
  });

  it('should create the component', () => {
    expect(component).toBeTruthy();
  });

  it('should match the snapshot with no document provided', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should match the snapshot with a pdf document provided', () => {
    conponentRef.setInput('document', <Document>{ extension: '.pdf' });
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
