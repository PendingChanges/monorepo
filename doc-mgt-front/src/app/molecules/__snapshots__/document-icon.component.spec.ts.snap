// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentIconComponent should match the snapshot with a pdf document provided 1`] = `
<app-document-icon
  document={[Function Function]}
  iconUrl={[Function Function]}
>
  <p-image
    class="p-element block xl:block mx-auto border-round w-full"
    width="30"
  >
    <span
      class="p-image p-component"
    >
      <img
        src="/pdf.svg"
        width="30"
      />
    </span>
  </p-image>
</app-document-icon>
`;

exports[`DocumentIconComponent should match the snapshot with no document provided 1`] = `
<app-document-icon
  document={[Function Function]}
  iconUrl={[Function Function]}
>
  <p-image
    class="p-element block xl:block mx-auto border-round w-full"
    width="30"
  >
    <span
      class="p-image p-component"
    >
      <img
        src="/unknown.svg"
        width="30"
      />
    </span>
  </p-image>
</app-document-icon>
`;
